File: back/src/middlewares/verifyToken.ts
import { Request, Response, NextFunction } from "express";
import jwt from "jsonwebtoken";

import dotenv from "dotenv";
dotenv.config();

const verifyToken = async (req: Request, res: Response, next: NextFunction) => {
  try {
    if (!req.headers["authorization"]) {
      res.status(401).json({
        status: "fail",
        message: "token이 없습니다",
      });
    } else {
      const secretKey: string = process.env.JWT_SECRET_KEY;
      const userToken = req.headers["authorization"].split(" ")[1];
      const jwtDecoded: any = jwt.verify(userToken, secretKey);
      req.currentUserId = jwtDecoded.userId;
      next();
    }
  } catch (error) {
    res.status(401).json({
      status: "fail",
      message: "token이 변형되었습니다. ",
      error,
    });
    next(error);
  }
};

export { verifyToken };

File: ai/verifyToken.py
import jwt
from functools import wraps
from flask import request, Response, g

import os 
from dotenv import load_dotenv
load_dotenv()


def verifyToken(f):      
    @wraps(f)                   
    def decorated_function(*args, **kwargs):
        header_token = request.headers.get('authorization')

        if header_token is not None:  
            accessToken = header_token.split(" ")[1]
            try:
                payload = jwt.decode(accessToken, os.environ.get('JWT_SECRET_KEY'), 'HS256') 
            except jwt.InvalidTokenError:
                payload = None     

            if payload is None: return Response(status=401)  

            userId = payload['userId']  
            g.userId = userId
        else:
            g.userId = None

        return f(*args, **kwargs)
    return decorated_function